### ISSUE TITLE
Android build suddenly stops/crashes in real device

### ISSUE THREAD

**Issue by mrinal-roy** [react-native-maps/react-native-maps#3477]
I have included MapView component in my app. The iOS build works fine with ios simulator as well as iphone real device.
However, whenever I work in Android emulator or I create a build for real android device, the app closes on going to the screen where the below component is rendered (sometimes it works fine, but after few times rendering of the component the app crashes/closes). No error message is shown on the screen of the emulator or on the real android phone.

```
<MapView
                        style={styles.locationMapWrapper}
                        region={this.state.region}
                        // initialRegion={this.state.region}
                        provider={PROVIDER_GOOGLE}
                        // mapType="standard"
                        showsScale
                        showsCompass
                        showsPointsOfInterest
                        // showsBuildings
                        liteMode={true}
                        // zoomEnabled={true}
                        onLayout={this.onMapLayOut}
                    >

                            <Marker
                                pinColor="red"
                                key={this.state.placesDetails.place_id}
                                coordinate={{
                                    latitude: parseFloat(this.state.placesDetails.restroom_data.lat),
                                    longitude: parseFloat(this.state.placesDetails.restroom_data.lng)
                                }}
                                ref = {ref => { this.marker = ref; }}
                                title={this.state.placesDetails.restroom_data.name}
                            >
                                {Platform.OS === "ios" ? (
                                    <Callout
                                        tooltip={true}
                                        style={styles.callout}
                                    >
                                        <Text style={styles.title}>
                                            {this.state.placesDetails.restroom_data.name}
                                        </Text>
                                    </Callout>
                                    ): null }
                            </Marker>
```

I am using RN 0.61.5 with react-native-maps: "^0.26.1". What could be wrong? Can somebody help me please.

**Comment by kub1x** [comment]
Exactly the same in Managed Expo app. Whole app crashes (and reloads as Expo error handling always does).  Even pure `return <MapView />;` component crashes the whole app. I thing it is not Javascript error as I've wrapped the JSX part with `try-catch` and nothing got caught. App just restarted. 

I have tried this tutorial for adding API KEY: https://medium.com/nycdev/create-a-react-native-app-with-google-map-using-expo-io-68041252023d didn't help. 

Expo SDK v37.0.1, react-native-maps 0.26.1, (tried also react-native-map-clustering 3.3.9, but it is irrelevant as it fails the same way with or without it).

**Comment by kub1x** [comment]
Hey, has anyone seen this issue, yet? 
I'd do what I can to help to solve this as it is a blocker for us. 
It happens the same in the newest Expo release v38 (RN 0.62) https://blog.expo.io/expo-sdk-38-is-now-available-ab6cd30ca2ee
Thanks in advance.

**Comment by kub1x** [comment]
Tried to create minimal failing repro: https://github.com/kub1x/expo-maps-repro
Created using: 
```
expo init # creates expo project with Expo SDK v38, React Native v0.63
expo install react-native-maps # just calls yarn add on fixed version 0.27.1
# here i made MapView the root app component
expo build:android --type apk
```
Running using `expo start` works. 
After downloading and installing apk to phone, it does not work (app hangs on splash screen). 
[Same source when started in Expo client](https://expo.io/@cedule/maps-repro) works (shows the map).

**Comment by kub1x** [comment]
Hey, so in my case the crash was caused by missing `android.config.googleMaps.apiKey` in Expo's `app.json` config file. Guess Android App Manifest is generated from it. `react-native-maps` should check the missing GOOGLE MAPS API KEY and fail gracefully (showing an error overlay over the map component or something) rather than bring down the whole app. More info in the expo bug here: https://github.com/expo/expo/issues/9136
